src/main/java/io/sdkman/broker/download/CandidateDownloadHandler.java

```diff
@@ -47,12 +58,44 @@ public void handle(Context ctx) {
                                             downloadResolver.resolve(downloads, p.name())
                                                     .ifPresentOrElse(v -> {
                                                         audit(details, p.id(), v.platform());
+                                                        writeChecksumHeaders(ctx, v.checksums());
                                                         ctx.redirect(302, v.url());
                                                     }, clientError(ctx, 404))),
                             clientError(ctx, 400));
         }, clientError(ctx, 404));
     }
 
+    private void writeChecksumHeaders(Context ctx,
+        Option<scala.collection.immutable.Map<String, String>> checksums) {
+        if(checksums == null || checksums.isEmpty()){
+            return;
+        }
+
+       Comparator<String> algoComparator =  Comparator.comparing((Function<String, Integer>) algorithm -> {
+            if (MD5.id().equals(algorithm)) {
+                return MD5.priority();
+            } else if (SHA1.id().equals(algorithm)) {
+                return SHA1.priority();
+            } else if (SHA224.id().equals(algorithm)) {
+                return SHA224.priority();
+            } else if (SHA256.id().equals(algorithm)) {
+                return SHA256.priority();
+            } else if (SHA384.id().equals(algorithm)) {
+                return SHA384.priority();
+            } else if (SHA512.id().equals(algorithm)) {
+                return SHA512.priority();
+            } else {
+                return 0;
+            }
+        }).reversed();
+
+        Map<String, String> sortedChecksums = new TreeMap<>(algoComparator);
+        sortedChecksums.putAll(JavaConverters.mapAsJavaMap(checksums.get()));
```

Comment:
It also means that you don't need to commit that heinous sin of calling `.get()` :smile:

Reply:
👍